[NO-JIRA] Infinite scroll docs

tiagoengel opened the pull request:
![0 0 0 0_8080_components_infinite-scroll_platform web 1](https://user-images.githubusercontent.com/1457263/41292758-6442ecd2-6e4b-11e8-824a-53aa42fc61f9.png)

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+const List = ({ elements }) => (
+  <div className={getClassName('bpk-docs-infinite-scroll-page__list')}>
+    {elements.map(element => <div key={element}>{element}</div>)}
+  </div>
```
It might be better to dogfood our own components and use `BpkList` and `BpkListItem` in the examples here.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+const components = [
+  {
+    id: 'infinite',
+    title: 'Infinite list',
```
This should probably be `default`.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+  {
+    id: 'infinite',
+    title: 'Infinite list',
+    blurb: [],
```
I think the examples on this page could use some blurb, as it's not entirely clear from the page what they do. @jamesf3rguson may have thoughts on what to put, but I was thinking something along the lines of:

> Loads five more list items upon reaching the bottom of the list.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+    ],
+  },
+  {
+    id: 'infiniteStop',
```
Our convention here is to use kebab case, e.g. `infinite-stop`.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+  {
+    id: 'infinite1',
+    title: 'One element per scroll',
+    blurb: [],
```
Blurb here would be good too I think, something like:

> Loads one more list item upon reaching the bottom of the list.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+  },
+  {
+    id: 'infiniteStop',
+    title: 'Partial load',
```
I think a blurb is needed here, something like:

> After 15 items have loaded, a button must be pressed to load more items.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,198 @@
+  },
+  {
+    id: 'infiniteCustomLoading',
+    title: 'Using a custom loadingComponent',
```
_Using a custom loading component_ would be better title, we want to keep it in plain English.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss:
```diff
@@ -0,0 +1,50 @@
+
+.bpk-docs-infinite-scroll-page {
+  &__fixed-panel {
+    height: 300px; /* stylelint-disable-line unit-blacklist */
```
We should use a token here, just take one of the existing ones and multiply it to make something of a similar height. E.g. `$bpk-spacing-lg * 10`

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.scss:
```diff
@@ -0,0 +1,50 @@
+
+  &__list {
+    div {
+      height: 4rem;
```
We should use a token here, just take one of the existing ones and multiply it to make something of a similar height.

shaundon commented on packages/bpk-docs/src/constants/routes.js:
```diff
@@ -100,6 +100,7 @@ export const TOUCHABLE_NATIVE_FEEDBACK =
   '/components/touchable-native-feedback';
 export const MAP = '/components/map';
 export const CAROUSEL = '/components/carousel';
+export const INFINITE_SCROLL = '/components/infiniteScroll';
```
This should be in kebab case to match convention, e.g. we have `/components/touchable-native-feedback`

tiagoengel commented:
Addressed all feedback @shaundon and @matthewdavidson. @jamesf3rguson had looked at it before (in the original PR) but I'll ask for a new design review.

shaundon commented on packages/bpk-docs/src/pages/InfiniteScrollPage/InfiniteScrollPage.js:
```diff
@@ -0,0 +1,202 @@
+
+const blurb = [
+  <IntroBlurb>
+    This High-order component is used to wrap a list component and add the
```
This should say `higher-order component`, not `High-order component`.

Sorry - I meant to add this to my original review and forgot.
